File: backend/CaseManagementSystem.Api/Models/Case.cs
namespace CaseManagementSystem.Api.Models;

public class Case
{
    public int Id { get; set; }
    public string Title { get; set; } = "";
    public string Description { get; set; } = "";
    public string Status { get; set; } = "New";
    public string Priority { get; set; } = "Medium";
    public string Category { get; set; } = "Other";

    public int CustomerId { get; set; }
    public Customer? Customer { get; set; }

    public int? AssignedUserId { get; set; }
    public User? AssignedUser { get; set; }

    public int CreatedByUserId { get; set; }
    public User? CreatedByUser { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? DueDate { get; set; }

    public List<Comment> Comments { get; set; } = new();
}

File: frontend/case-management-system-ui/src/types/case.ts
export interface CaseListItem {
  id: number;
  title: string;
  status: string;
  priority: string;
  category: string;
  customerName: string;
  assignedUserName: string;
  createdAt: string;
  updatedAt: string;
  dueDate: string | null;
}

export interface CaseComment {
  id: number;
  userId: number;
  userName: string;
  content: string;
  createdAt: string;
}

export interface CaseDetails {
  id: number;
  title: string;
  description: string;
  status: string;
  priority: string;
  category: string;
  customerId: number;
  customerName: string;
  customerEmail: string;
  customerCompanyName: string;
  customerPhoneNumber: string;
  assignedUserId: number | null;
  assignedUserName: string;
  createdByUserId: number;
  createdByUserName: string;
  createdAt: string;
  updatedAt: string;
  dueDate: string | null;
  comments: CaseComment[];
}

export interface CaseQueryParams {
  status?: string;
  priority?: string;
  category?: string;
  assignedUserId?: number;
  search?: string;
}

export interface UpdateCaseRequest {
  title: string;
  description: string;
  status: string;
  priority: string;
  category: string;
  customerId: number;
  assignedUserId: number | null;
  dueDate: string | null;
}

export interface AddCommentRequest {
  userId: number;
  content: string;
}

export interface CreateCaseRequest {
  title: string;
  description: string;
  status: string;
  priority: string;
  category: string;
  customerId: number;
  assignedUserId: number | null;
  createdByUserId: number;
  dueDate: string | null;
}